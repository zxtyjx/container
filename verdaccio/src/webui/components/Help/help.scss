.help {
  .noPkg {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 30px 0;
    font-size: 20px;
    color: #9f9f9f;

    .noPkgTitle {
      margin: 1em 0;
      padding: 0;
      font-size: 24px;
    }

    .noPkgIntro {
      line-height: 1.5;
      margin: 0 auto;
      font-size: 14px;
    }

    code {
      font-style: italic;
    }
  }
}
