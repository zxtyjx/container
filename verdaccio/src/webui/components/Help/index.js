import React from 'react';
import SyntaxHighlighter, {registerLanguage} from 'react-syntax-highlighter/dist/light';
import sunburst from 'react-syntax-highlighter/src/styles/sunburst';
import js from 'react-syntax-highlighter/dist/languages/javascript';

import classes from './help.scss';
import {getRegistryURL} from '../../utils/url';

registerLanguage('javascript', js);

const Help = () => {
  const registryURL = getRegistryURL();

    return (
      <div className={classes.help}>
        <li className={classes.noPkg}>
          <h1 className={classes.noPkgTitle}>
            No Package Published Yet
          </h1>
          <div className={classes.noPkgIntro}>
            <div>
              To publish your first package just:
            </div>
            <br/>
            <strong>
              1. Login
            </strong>
            <SyntaxHighlighter language='javascript' style={sunburst} id="adduser">
              {`npm adduser --registry  ${registryURL}`}
            </SyntaxHighlighter>
            <strong>2. Publish</strong>
            <SyntaxHighlighter language='javascript' style={sunburst} id="publish">
              {`npm publish --registry ${registryURL}`}
            </SyntaxHighlighter>
            <strong>3. Refresh this page!</strong>
          </div>
        </li>
      </div>
    );
};

export default Help;
