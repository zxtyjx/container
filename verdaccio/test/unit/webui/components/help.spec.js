/**
 * Help component
 */
import React from 'react';
import { shallow } from 'enzyme';
import SyntaxHighlighter from 'react-syntax-highlighter/dist/light';
import Help from '../../../../src/webui/components/Help/index';

describe('<Help /> component', () => {

  it('should set html from props with / base path', () => {
    jsdom.reconfigure({
      url: "http://example.com/"
    });
    const wrapper = shallow(<Help />);
    expect(
      wrapper
        .find('#adduser')
        .find(SyntaxHighlighter)
        .dive()
        .text()
    ).toEqual('npm adduser --registry  http://example.com');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should set html from props with someOtherPath', () => {
    jsdom.reconfigure({
      url: "http://example.com/someOtherPath"
    });
    const wrapper = shallow(<Help />);
    expect(
      wrapper
        .find('#publish')
        .find(SyntaxHighlighter)
        .dive()
        .text()
    ).toEqual('npm publish --registry http://example.com/someOtherPath');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
