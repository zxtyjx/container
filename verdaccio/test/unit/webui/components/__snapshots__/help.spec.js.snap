// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Help /> component should set html from props with / base path 1`] = `"<div class=\\"help\\"><li class=\\"noPkg\\"><h1 class=\\"noPkgTitle\\">No Package Published Yet</h1><div class=\\"noPkgIntro\\"><div>To publish your first package just:</div><br/><strong>1. Login</strong><pre id=\\"adduser\\" style=\\"display:block;overflow-x:auto;padding:0.5em;background:#000;color:#f8f8f8\\"><code>npm adduser --registry  http:<span style=\\"color:#aeaeae;font-style:italic\\">//example.com</span></code></pre><strong>2. Publish</strong><pre id=\\"publish\\" style=\\"display:block;overflow-x:auto;padding:0.5em;background:#000;color:#f8f8f8\\"><code>npm publish --registry http:<span style=\\"color:#aeaeae;font-style:italic\\">//example.com</span></code></pre><strong>3. Refresh this page!</strong></div></li></div>"`;

exports[`<Help /> component should set html from props with someOtherPath 1`] = `"<div class=\\"help\\"><li class=\\"noPkg\\"><h1 class=\\"noPkgTitle\\">No Package Published Yet</h1><div class=\\"noPkgIntro\\"><div>To publish your first package just:</div><br/><strong>1. Login</strong><pre id=\\"adduser\\" style=\\"display:block;overflow-x:auto;padding:0.5em;background:#000;color:#f8f8f8\\"><code>npm adduser --registry  http:<span style=\\"color:#aeaeae;font-style:italic\\">//example.com/someOtherPath</span></code></pre><strong>2. Publish</strong><pre id=\\"publish\\" style=\\"display:block;overflow-x:auto;padding:0.5em;background:#000;color:#f8f8f8\\"><code>npm publish --registry http:<span style=\\"color:#aeaeae;font-style:italic\\">//example.com/someOtherPath</span></code></pre><strong>3. Refresh this page!</strong></div></li></div>"`;
